### PR TITLE
feat(DENG-1774): adding fenix derived firefox android clients v2

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(DENG-1774): adding fenix derived firefox android clients v2

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1775)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+      CASE
+        WHEN first_session.adjust_info IS NOT NULL
+          THEN "first_session"
+        WHEN metrics.adjust_info IS NOT NULL
+          THEN "metrics"
+        ELSE NULL
+      END AS adjust_info__source_ping
+    ) AS metadata,
+    -- field to help us identify suspicious devices on iOS, for more info see: bug-1846554
+    (app_version = '107.2' AND submission_date >= '2023-02-01') AS is_suspicious_device_client,
```
Comment (bani) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql:149]: this shouldn't be here, the incident is iOS only

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      _previous.adjust_creative,
+      _previous.adjust_network
+    )
+  ).*,
+  STRUCT(
+    COALESCE(
+      _previous.metadata.is_reported_first_session_ping
+      OR _current.metadata.is_reported_first_session_ping,
+      FALSE
+    ) AS is_reported_first_session_ping,
```
Comment (bani) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql:205]: current name on Android is just `reported_first_session_ping`. We can change it for consistency, but this field is used a lot so it'll break a bunch of queries once we update the view.

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql (hunk 1 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql:149]: oh yes, fair point. I thought I deleted this.

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql (hunk 2 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql]: ok, changed the naming for android.

**Review by bani — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/init.sql [hunk 3]
```diff
@@ -0,0 +1,183 @@
+-- Query first observations for Firefox Android Clients.
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/init.sql:1]: Let's remove the `init.sql` file in favor of `is_init()` in the `query.sql` file

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  COALESCE(
+    _previous.first_reported_country,
+    _current.first_reported_country
+  ) AS first_reported_country,
+  COALESCE(_previous.first_reported_isp, _current.first_reported_isp) AS first_reported_isp,
+  COALESCE(_previous.channel, _current.channel) AS channel,
+  COALESCE(_previous.device_manufacturer, _current.device_manufacturer) AS device_manufacturer,
+  COALESCE(_previous.device_model, _current.device_model) AS device_model,
+  COALESCE(_previous.os_version, _current.os_version) AS os_version,
+  COALESCE(_previous.app_version, _current.app_version) AS app_version,
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql:180]: Can you confirm this isn't missing any fields from `firefox_android_clients`? (sans `last_reported` fields, we can drop those). I'm seeing e.g. locale missing.

**Review by fbertsch — CHANGES_REQUESTED** [review]
Before we implement this change, I think we need to have a discussion with Adjust. When we were working on client-id regeneration, they indicated to me that attribution _can_ change. IIRC they said it may go from missing to _something_. If that "missing" is NULLs or empty strings, this change will work as we expect. However it they fill in even a single value then we won't get the updated attribution later.

**Comment by fbertsch** [comment]
@kik-kik I sent them an email and cc'ed you.

**Comment by kik-kik** [comment]
Hey @fbertsch , sorry I've been out on PTO. What is the conclusion, apart from using `is_init()` and adding `locale` are we good?

**Comment by kik-kik** [comment]
@fbertsch `locale` is now in the query and the query has been modified to support `is_init()`. Could you please re-review the PR? 🙇

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql (hunk 4 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql:202]: As far as I can see they are all there, locale should be there now too.

**Review by fbertsch — APPROVED** [review]
LGTM. Adjust has indicated that we will only get a different attribution if a user is [reattributed](https://help.adjust.com/en/article/reattribution), which would happen if they become dormant. That's not a situation we optimize for in marketing.

**Comment by kik-kik** [comment]
> LGTM. Adjust has indicated that we will only get a different attribution if a user is [reattributed](https://help.adjust.com/en/article/reattribution), which would happen if they become dormant. That's not a situation we optimize for in marketing.

Thanks for double checking @fbertsch 🙇.

Will merge the change now, and deploy.

To infinity and beyond! 🚀

**Pull request merged**